research/src/pages/radio-button/radio-button.proposal.mdx

```diff
@@ -0,0 +1,185 @@
+---
+menu: Proposals
+name: Radio Button (Editor's Draft)
+path: /components/radio-button
+pathToResearch: /components/radio-button.research
+---
+
+import RadioButtonAnatomy from '../../components/radio-button-anatomy'
+import RadioButtonGroupAnatomy from '../../components/radio-button-group-anatomy'
+
+## Overview
+
+The `<oui-radio-button>` element is a control that allows the user to select a single choice out of a set of choices. It has a binary "checked" state, either checked or unchecked (where "checked" is the selected choice, while all the other remain "unchecked".
+
+Because the `<oui-radio-button>` is a choice from a set of choices, the choices are logically grouped by an implicit or explicit `<oui-radio-button-group>`. This proposal will deal with the existence of both elements, the different ways they can interact and their properties.
+
+### Use Cases
+
+The `<oui-radio-button-group>` is used in cases very similar to a `<select>` element, where the user is presented with a set of options from which they can select a single one.
+
+Differently from `<select>`, where all the `<option>` are condensed into a single box, the `<oui-radio-button-group>` has all the `<oui-radio-button>` visible, readily available to be read and selected. Choosing between a `<select>` and `<oui-radio-button-group>` is a matter of layout and information density preference, in most cases.
+
+Also differently from a `<select>`, there is currently no `multiple` attribute to allow a multiple choice group, and this document will not propose one for now.
+
+## Prior Art/Examples
+
+- [Radio input](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/radio)
+- [WAI-ARIA Radio](https://www.w3.org/TR/wai-aria-practices/#radiobutton)
+- [WHATWG](<https://html.spec.whatwg.org/multipage/input.html#radio-button-state-(type=radio)>)
+
+[Further examples on research](/components/radio-button.research).
+
+## Radio Button Design
+
+This section relates to a single `<oui-radio-button>`. Because it cannot stand on its own, without being part of and either explicit or implicit `<oui-radio-button-group>`, it does not have attributes such `required` or `readonly`, those being attributes of the group. A single `required` option is a group makes the other choices pointless, while a `readonly` option makes itself pointles.
```

Comment:
re: the allowance of the required attribute on a single radio - this makes sense in the context of a `radiogroup` where the only children of the group would be expected to be radio buttons, and at the parent level the requiredness of the grouping could be declared. 

However, current radio buttons in HTML have greater flexibility in where they can be placed / associated with each other, and can still be communicated as required with this flexibility.

for example:

```
<label><input type=radio name=r required> Choice 1</label>
Other information or form fields, or both, associated with making choice 1.
...
<label><input type=radio name=r> Choice 2</label>
Other information or form fields, or both, associated with making choice 2.
```

This would need to be thought out more in how to communicate the required nature of radios that don't neatly fit into a grouping (see my next comment for more on that) or a merging of what HTML presently allows (putting the required attribute on a radio button) and what the ARIA radio/radiogroup roles expect (that if you're using radios - particularly when a choice is required - they are part of a radiogroup, is more limiting than HTML)

Reply:
Does the code you used as an example mean that, by placing `required` in one of the options, the **group** is required?

Reply:
yes. the implicit grouping becomes required and you cannot submit a form unless one of the radios is checked

Reply:
Good to know, I was not aware of it.